abstract_specification.md

```diff
@@ -18,6 +18,15 @@ SOMA is an abstract _API specification_, with the goal of enabling multiple conc
 
 This document is attempting to codify the abstract, language-neutral SOMA data model and functional operations. Other specifications will document specific language bindings and particular storage system implementations. Where the term `language-specific SOMA specification` is used below, it implies a specification of the SOMA API as it is presented in a given language or computing environment (e.g., the SOMA Python API), common across all storage engine implementations in that language.
 
+# Lifecycle Stages
+
+The SOMA API uses [RStudio's lifecycle stage model](https://lifecycle.r-lib.org) to indicate the maturity of its classes and methods. The lifecycle stages are:
+
+- `experimental`: Under active development and may undergo significant and breaking changes.
```

Comment:
Worth adding something like "If an interface is unmarked, it should be assumed to be experimental."

Reply:
Great suggestion. I'll spell out that experimental is currently the default.